### ISSUE TITLE
clang static analyzer: dividezero checker warning on unreachable line

### ISSUE THREAD

**Issue by kangeroo-gong** [llvm/llvm-project#56801]
For the following program:
```c++
#include "map"

using namespace std;
class Qes {
private:
    map<int, int> map;
    int num;
public:
    void func(int a);
};

void Qes::func(int a) {
    if (a==num)
        return;
    auto iter = map.find(a); // no warning without this line
    if (a!=num)
        return;
    int c = 10/(a-num); // this line warn Division by zero
}
```
There are two if stmt make program can't reach  `int c = 10/(a-num)` ,but the checker still warn..

(Tested with **LLVM 14.0.6.**)

**Comment by EugeneZelenko** [comment]
Could you please try `main` branch?

**Comment by kangeroo-gong** [comment]
> Could you please try `main` branch? https://godbolt.org should be helpful.

<img width="1778" alt="image" src="https://user-images.githubusercontent.com/22929765/182064205-33599fc3-3d66-436d-9202-bb6e614cf732.png">
truck get same result

**Comment by kangeroo-gong** [comment]
https://discourse.llvm.org/t/false-positives-of-clang-static-analyzer-divzerochecker-in-llvm14-0-6/64146/4
haoNoQ have replied this issue in discourse
